### PR TITLE
Created data source `aws_api_gateway_resource`

### PR DESCRIPTION
**Pull request by BSick7**
Changes proposed in this pull request:

* Create `data.aws_api_gateway_resource`.

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccDataSourceAwsApiGatewayResource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestDataSourceAwsApiGatewayResource -timeout 120m
?       github.com/terraform-providers/terraform-provider-aws   [no test files]
=== RUN   TestDataSourceAwsApiGatewayResource
--- PASS: TestDataSourceAwsApiGatewayResource (17.08s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       (cached)

```

### CODE REVIEW

File: aws/data_source_aws_api_gateway_resource.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func dataSourceAwsApiGatewayResourceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	restApiId := d.Get("rest_api_id").(string)
+	target := d.Get("path").(string)
+	params := &apigateway.GetResourcesInput{RestApiId: aws.String(restApiId)}
+
+	var matches []*apigateway.Resource
```
Comment (bflad) [inline aws/data_source_aws_api_gateway_resource.go]: Is it possible to have multiple resources at the same path? If not, we should change this to be singular and `return false` in the `GetResourcesPages` after it matches to save API calls.

File: aws/data_source_aws_api_gateway_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceAwsApiGatewayResourceCheck("aws_api_gateway_resource.example_v1"),
+					testAccDataSourceAwsApiGatewayResourceCheck("aws_api_gateway_resource.example_v1_endpoint"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAwsApiGatewayResourceCheck(name string) resource.TestCheckFunc {
```
Comment (bflad) [inline aws/data_source_aws_api_gateway_resource_test.go]: We can skip creating this additional function by using the [`resource.TestCheckResourceAttrPair()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrPair) helper function, e.g.

```go
resourceName := "aws_api_gateway_resource.example_v1"
dataSourceName := "data.aws_api_gateway_resource.example_v1"
// ...
resource.TestCheckResourceAttrPair(resourceName, "id", dataSourceName, "id"),
resource.TestCheckResourceAttrPair(resourceName, "parent_id", dataSourceName, "parent_id"),
resource.TestCheckResourceAttrPair(resourceName, "path_part", dataSourceName, "path_part"),
```

File: aws/data_source_aws_api_gateway_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestDataSourceAwsApiGatewayResource(t *testing.T) {
```
Comment (bflad) [inline aws/data_source_aws_api_gateway_resource_test.go]: For acceptance tests the convention is to prefix them with `TestAcc` 👍

File: aws/data_source_aws_api_gateway_resource.go (hunk 1 above)
Comment (BSick7) [reply aws/data_source_aws_api_gateway_resource.go]: Good idea.
Changing now.

**Comment by BSick7** [comment]
@bflad I updated based on your great comments. I reverified acceptance tests pass.

**Comment by BSick7** [comment]
Super fast! Thanks @bflad!

**Comment by bflad** [comment]
This has been released in [version 1.33.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1330-august-22-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**